src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -3219,10 +3219,6 @@ module internal ParseAndCheckFile =
 
             use _unwindBP = UseBuildPhase BuildPhase.TypeCheck
 
-            // Apply nowarns to tcConfig (may generate errors, so ensure diagnosticsLogger is installed)
-            let tcConfig =
-                ApplyNoWarnsToTcConfig(tcConfig, parsedMainInput, !! Path.GetDirectoryName(mainInputFileName))
-
             // update the error handler with the modified tcConfig
```

Comment:
For my understanding, why this had to be removed (and nothing equivalent readded then)?

Reply:
The nowarn information (now: warn scope information) is no longer threaded from post-parse (where  it is created) to the type checking diagnostics logger by means of hiding it in the AST (where IMO it does not belong), but is stored now directly in the diagnostic options. See also the more detailed comments below.

(BTW I think the error handler update in the next line can also be removed. I will look into that.)

Reply:
Nice, thanks for the explanation!